src/main/java/org/truffleruby/core/module/ModuleNodes.java

```diff
@@ -2584,4 +2584,22 @@ protected Object doClass(RubyClass rubyClass) {
             return rubyClass.isSingleton;
         }
     }
+
+    @CoreMethod(names = "undefined_instance_methods")
+    public abstract static class UndefinedInstanceMethodsNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        @TruffleBoundary
+        protected RubyArray undefinedInstanceMethods(RubyModule module) {
+            List<RubySymbol> methodNames = new ArrayList<>();
+
+            for (InternalMethod methodEntry : module.fields.getMethods()) {
+                if (methodEntry != null && methodEntry.isUndefined()) {
```

Comment:
**siggestion:** It seems `methodEntry != null` is redundant. As far as I see  `module.fields.getMethods()` isn't supposed to return `null` elements and it would be better to fails explicitly instead of silently ignore the issue.

Reply:
Ah, good to know! Updating.